spec.md

```diff
@@ -3,193 +3,138 @@ title: "HTTP API V2"
+   - [Base](#base)
+   - [Tags](#tags)
+   - [Manifest](#manifest)
+      - [GET Manifest](#get-manifest)
+      - [PUT Manifest](#put-manifest)
+      - [DELETE Manifest](#delete-manifest)
+   - [Blob](#blob)
+      - [GET Blob](#get-blob)
+         - [Fetch Blob](#fetch-blob)
+         - [Fetch Blob Part](#fetch-blob-part)
+      - [DELETE Blob](#delete-blob)
+   - [Initiate Blob Upload](#initiate-blob-upload)
+      - [POST Initiate Blob Upload](#post-initiate-blob-upload)
+         - [Initiate Monolithic Blob Upload](#initiate-monolithic-blob-upload)
+         - [Initiate Resumable Blob Upload](#initiate-resumable-blob-upload)
+         - [Mount Blob](#mount-blob)
+   - [Blob Upload](#blob-upload)
+      - [GET Blob Upload](#get-blob-upload)
+      - [PATCH Blob Upload](#patch-blob-upload)
+      - [PUT Blob Upload](#put-blob-upload)
+      - [DELETE Blob Upload](#delete-blob-upload)
+   - [Catalog](#catalog)
+      - [GET Catalog](#get-catalog)
+
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard][c99-unspecified].
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
+An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
+
+## Historical Context
+
+For relevant details and a history leading up to this specification, please see the following issues:
 
 - [docker/docker#8093](https://github.com/docker/docker/issues/8093)
 - [docker/docker#9015](https://github.com/docker/docker/issues/9015)
 - [docker/docker-registry#612](https://github.com/docker/docker-registry/issues/612)
 
-### Scope
+## Scope
 
-This specification covers the URL layout and protocols of the interaction between docker registry and docker core.
-This will affect the docker core registry API and the rewrite of docker-registry.
-Docker registry implementations may implement other API endpoints, but they are not covered by this specification.
+This specification covers URL layout and protocols for interaction between a registry and registry client.
+Registry implementations MAY implement other API endpoints, but they are not covered by this specification.
 
-This includes the following features:
+This specification includes the following features:
 
 - Namespace-oriented URI Layout
 - PUSH/PULL registry server for V2 image manifest format
 - Resumable layer PUSH support
 - V2 Client library implementation
```

Comment:
This last item does not belong as part of `This specification`

Reply:
I know we said no to the registry daemon. Though it was up in the air for enough of client side for basic validation.

Reply:
Changing to: "- V2 Client (Consumer) requirements"

Reply:
Ok, let's not touch this and discuss in #23